jdaviz/configs/cubeviz/plugins/moment_maps/moment_maps.py

```diff
@@ -126,6 +148,14 @@ def calculate_moment(self, add_data=True):
         if data_wcs:
             data_wcs = data_wcs.swapaxes(0, 1)  # We also transpose WCS to match.
         self.moment = CCDData(analysis.moment(slab, order=n_moment).T, wcs=data_wcs)
+        if self.output_unit == "Velocity":
```

Comment:
```suggestion
        if n_moment > 0 and self.output_unit == "Velocity":
```

Reply:
Ah right, I didn't actually add any handling to switch back to `Wavelength` if you go back to moment 0. Good catch.